client/src/featureform/serving.py

```diff
@@ -508,7 +508,12 @@ def feature_df_from_transformation(self, feature):
         if isinstance(df, pd.Series):
             df = df.to_frame()
             df.reset_index(inplace=True)
-        if feature["source_timestamp"] != "":
+        if feature["source_timestamp"] != "" and feature["source_timestamp"] not in df:
+            raise ValueError(
+                f"Provided timestamp column {feature['source_timestamp']} not found in feature {feature['name']}; "
+                f"either remove 'timestamp_column' from the feature registration or include it in the transformation."
```

Comment:
My assumption would be that a feature would most likely be build of a transformation. But we would have cases where it uses the primary source. Should we modify this statement to say include it in the source.